plugins/modules/content_export_info.py

```diff
@@ -130,35 +122,28 @@
 
 '''
 
-from ansible_collections.theforeman.foreman.plugins.module_utils.foreman_helper import KatelloAnsibleModule, _flatten_entity
+from ansible_collections.theforeman.foreman.plugins.module_utils.foreman_helper import KatelloInfoAnsibleModule
 
 
-class KatelloContentExportInfoModule(KatelloAnsibleModule):
+class KatelloContentExportInfo(KatelloInfoAnsibleModule):
     pass
 
 
 def main():
-    module = KatelloContentExportInfoModule(
+    module = KatelloContentExportInfo(
         foreman_spec=dict(
             id=dict(required=False, type='int'),
-            content_view_version_id=dict(required=False, type='int'),
-            content_view_id=dict(required=False, type='int'),
+            content_view_version=dict(type='entity', scope=['content_view'], required=False),
+            content_view=dict(type='entity', scope=['organization'], required=False),
             destination_server=dict(required=False, type='str'),
-            organization_id=dict(required=False, type='int'),
             type=dict(required=False, type='str', choices=['complete', 'incremental']),
             search=dict(required=False, type='str'),
+            name=dict(invisible=True),
         ),
     )
 
     with module.api_connection():
-        module.auto_lookup_entities()
-
-        endpoint = 'content_exports'
```

Comment:
I still don't understand how it knows which endpoint to call if we delete this. :open_mouth:

Reply:
The class is called `KatelloContentExportInfo`, it strips "Katello" and "Info" and then translates the rest from "CamelCaseSingular" to "snake_case_plural".

https://github.com/theforeman/foreman-ansible-modules/blob/e27433353a67f65e5c0f845fd1d6e60eade77651/plugins/module_utils/foreman_helper.py#L1349-L1369